improver/generate_ancillaries/generate_timezone_mask.py

```diff
@@ -348,8 +349,12 @@ def process(self, cube):
         # Create a cube containing the timezone UTC offset information.
         timezone_mask = iris.cube.CubeList()
         for offset in range(min_offset, max_offset + 1):
-            zone = (grid_offsets != offset).astype(np.int32)
-            coord = iris.coords.DimCoord([offset], long_name="UTC_offset")
+            zone = (grid_offsets != offset).astype(np.int8)
+            coord = iris.coords.DimCoord(
+                np.array([offset], dtype=np.int32),
+                long_name="UTC_offset",
+                units="hours",
```

Comment:
There are a few political entities that have non-unit-hours UTC offsets - do we ever need to worry about those?

Reply:
As mentioned already in this bit of code, we aren't producing data at a finer temporal resolution than one hour, so sub-hour time-zone offsets need rounding to a whole-hour or we won't find a match for them.